src/test/java/org/wildfly/security/sasl/oauth2/OAuth2SaslClientTest.java

```diff
@@ -158,8 +157,8 @@ public void failedWithBearerTokenFromConfiguration() throws Exception {
         try {
             message = saslServer.evaluateResponse(message);
         } catch (SaslException e) {
-            assertTrue(e.getCause() instanceof RealmUnavailableException);
-            RealmUnavailableException cause = (RealmUnavailableException) e.getCause();
+            assertTrue(e.getCause() instanceof IllegalArgumentException);
```

Comment:
What I mean however is this Exception is coming from the realm - the realm instead should be logging why evidence verification failed and returning false.

If we want an exception here the validation should really be in the mech.